src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> t = tail;
+        final Node<T> newNode = new Node<>(t,value,null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, false);
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, true);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, true);
+        Node<T> x = node(index);
+        T oldVal = x.item;
+        x.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, true);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> x = head;
+        for (int i = 0; i < size; i++) {
+            if (x.item == object || x.item.equals(object)) {
```

Comment:
```suggestion
            if (x.item == object || x.item != null && x.item.equals(object)) {
```
it's better to add one more check in case item == null and object != null